pkg/util/client/client.go

```diff
@@ -31,23 +31,33 @@ func CreatePatch(before, after client.Object) (client.Patch, error) {
 	return client.RawPatch(patchBase.Type(), patchBytes), nil
 }
 
-func Patch(ctx context.Context, c client.Client, before, after client.Object) error {
-	patch, err := CreatePatch(before, after)
+func Patch(ctx context.Context, c client.Client, obj client.Object, update func() (bool, error)) error {
+	objOriginal := obj.DeepCopyObject().(client.Object)
```

Comment:
actually, can you also do:

```golang
objOriginal.SetResourceVersion("")
```

This will give us the same behavior as the original code (to fail the patch if there was any other update that we are not aware of).

Reply:
Done